src/WebJobs.Script.WebHost/Middleware/SystemTraceMiddleware.cs

```diff
@@ -46,13 +46,22 @@ public async Task Invoke(HttpContext context)
 
             await _next.Invoke(context);
 
+            LogIfOffline(request, scriptHostManager.State);
             sw.Stop();
             details["identities"] = GetIdentities(context);
             details["status"] = context.Response.StatusCode;
             details["duration"] = sw.ElapsedMilliseconds;
             _logger.Log(LogLevel.Information, 0, logData, null, (s, e) => $"Executed HTTP request: {details}");
         }
 
+        private void LogIfOffline(HttpRequest request, ScriptHostState hostState)
+        {
+            if (request.IsAdminRequest() && hostState == ScriptHostState.Offline)
+            {
+                _logger.Log(LogLevel.Warning, new EventId(2, "HostIsOffline"), "Host is offline.");
```

Comment:
Host being offline during admin requests isn't a Warning condition, it's normal. So I don't recommend logging warnings for this.

Reply:
Ok, I thought [Host Unavailable](https://github.com/Azure/azure-functions-host/blob/0487f93c3ab6dda28c74f6a39b3cd6ddf5a262d9/src/WebJobs.Script.WebHost/Middleware/HostAvailabilityCheckMiddleware.Log.cs#L20) or offline was a warning condition. 

Makes sense though, if this is expected. I will update it.

Reply:
Yeah - host is taken offline to install extensions for example, and while offline, other admin apis are available for portal to call.

Reply:
Why don't we move this into the Ping controller method so it's just scoped to pings, and we can log a generalized Verbose log message with event ID "PingStatus" with json content that we can add more stuff to in the future as needed? Initially it'll just be `{ hostState: <state> }`. Then the ping becomes a generalized heartbeat log of sorts for use by detectors.

Reply:
I think that's a good call. Looks like https://github.com/Azure/azure-functions-host/blob/dev/src/WebJobs.Script.WebHost/Models/HostStatus.cs is already used for `/admin/host/status` calls. I can re-use that, although we probably don't need to log anything other than the `state` from that model.

Reply:
I wouldn't reuse HostStatus. Just log what you need to for ping.